src/test/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/integration/MigrationControllerIntTest.kt

```diff
@@ -443,6 +443,54 @@ class MigrationControllerIntTest : IntegrationTestBase() {
     }
   }
 
+  @Nested
+  inner class ListAllClientDetails {
+    @Test
+    fun `access unauthorized when no authority`() {
+      webTestClient.get().uri("/client-details")
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+      webTestClient.get().uri("/client-details")
+        .headers(setAuthorisation(roles = listOf()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access forbidden when wrong role`() {
+      webTestClient.get().uri("/client-details")
+        .headers(setAuthorisation(roles = listOf("WRONG")))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `list client details success`() {
+      val matchByClientId = "$[?(@.clientId == '%s')]"
+      webTestClient.get().uri("/client-details")
+        .headers(setAuthorisation(roles = listOf("ROLE_OAUTH_ADMIN")))
+        .exchange()
+        .expectStatus().isOk
+        .expectHeader().contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .jsonPath("$.[*]").value<List<String>> { assertThat(it).hasSize(10) }
+        .jsonPath(matchByClientId, "test-client-id").exists()
```

Comment:
Would be better if the test confirms that the expected number of records are present, and then confirms that all expected fields are present for one record